src/librustdoc/html/markdown.rs

```diff
@@ -1174,12 +1173,36 @@ crate fn markdown_links(md: &str) -> Vec<(String, Option<Range<usize>>)> {
             } else {
                 None
             }
-        };
+        }
```

Comment:
Wow this whole thing is just *so* concerning. You don't need to change it here but this should really be fixed to get rid of the unsafe, there's no need at all for it.

Reply:
Simple enough, if the goal is only to get rid of the unsafe block.

Also, since we only use this inside "new_with_source", which is in turn only used if we have a borrowed string from the parser, maybe I can just delete 3/4 of this function.

Reply:
I fixed this in https://github.com/rust-lang/rust/pull/80244.